### PR TITLE
[FLINK-30554][sql-client] Remove useless sessionId in the Executor

### PR DESCRIPTION
**Pull request by fsk119**
## What is the purpose of the change

*Remove useless sessionId in the Executor. The SQL Gateway uses the `SessionHandle` to identify each Session and the `SessionHandle` is assigned by the Gateway only. The `Executor` in the SqlClient is the client to submit messages to the 
Gateway and it is only used by the current Session. We don't need to use the sessionId everywhere.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by fsk119** [comment]
@flinkbot run azure

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java [hunk 1]
```diff
@@ -99,88 +94,69 @@ public void start() {
...
         context.reset(key);
     }
 
     @Override
-    public void setSessionProperty(String sessionId, String key, String value)
-            throws SqlExecutionException {
-        SessionContext context = getSessionContext(sessionId);
+    public void setSessionProperty(String key, String value) throws SqlExecutionException {
+        SessionContext context = sessionContext;
         context.set(key, value);
```
Comment (swuferhong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java:154]: ditto

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java [hunk 2]
```diff
@@ -85,16 +85,15 @@ private void start() {
             executor.start();
 
             // Open an new session
```
Comment (swuferhong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:87]: nit: We can correct these obvious mistakes together： `an` -> `a`.  Btw, you can also pay attention to the warning of the idea checkStyle, like this class.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java [hunk 3]
```diff
@@ -320,19 +294,17 @@ public void cancelQuery(String sessionId, String resultId) throws SqlExecutionEx
     }
 
     @Override
-    public void removeJar(String sessionId, String jarUrl) {
-        final SessionContext context = getSessionContext(sessionId);
+    public void removeJar(String jarUrl) {
+        final SessionContext context = sessionContext;
         context.removeJar(jarUrl);
```
Comment (swuferhong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java:299]: ditto

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java [hunk 4]
```diff
@@ -240,10 +215,9 @@ public TableResultInternal executeModifyOperations(
     }
 
     @Override
-    public ResultDescriptor executeQuery(String sessionId, QueryOperation query)
-            throws SqlExecutionException {
-        final TableResultInternal tableResult = executeOperation(sessionId, query);
-        final SessionContext context = getSessionContext(sessionId);
+    public ResultDescriptor executeQuery(QueryOperation query) throws SqlExecutionException {
+        final TableResultInternal tableResult = executeOperation(query);
+        final SessionContext context = sessionContext;
```
Comment (swuferhong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java:220]: ditto

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java [hunk 5]
```diff
@@ -519,43 +516,38 @@ public String openSession(@Nullable String sessionId) throws SqlExecutionExcepti
...
-            return this.sessionMap.get(sessionId).getConfigMap();
+        public Map<String, String> getSessionConfigMap() throws SqlExecutionException {
+            return sessionContext.getConfigMap();
         }
 
         @Override
-        public ReadableConfig getSessionConfig(String sessionId) throws SqlExecutionException {
-            SessionContext context = this.sessionMap.get(sessionId);
+        public ReadableConfig getSessionConfig() throws SqlExecutionException {
+            SessionContext context = sessionContext;
```
Comment (swuferhong) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliClientTest.java:533]: ditto

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java [hunk 6]
```diff
@@ -42,54 +42,50 @@ public interface Executor {
...
     /**
      * Returns a copy of {@link Map} of all session configurations that are defined by the executor
      * and the session.
      *
-     * <p>Both this method and {@link #getSessionConfig(String)} return the same configuration set,
-     * but different return type.
+     * <p>Both this method and {@link #getSessionConfig()} return the same configuration set, but
+     * different return type.
      */
-    Map<String, String> getSessionConfigMap(String sessionId) throws SqlExecutionException;
```
Comment (swuferhong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java:65]: I have one question. Original  `LocalExecutor` implements `Executor` have a map `ConcurrentHashMap<String, SessionContext> contextMap`, but now, we just delete it. Does this mean that each executor can only have one session? (If that, why we need sessionId, we can just define an executorId instead ? ). Also, I think if you delete this `contextMap`, you need add tests to verify we don't need this map. WDYT ？

**Review by swuferhong — COMMENTED** [review]
Thanks for the contribution @fsk119  , I left some comments

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java (hunk 6 above)
Comment (fsk119) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java:65]: > Does this mean that each executor can only have one session?
Yes, you are right. Executor only works for one session in the new design. The Gateway is designed for the multitenancy.

> why we need sessionId, we can just define an executorId instead?
I think session id is still useful. For example, SQL Client can convey the user name/password with the current API. So I am prone to keep it unchanged and prepare for the future authentication mechanism.

**Comment by fsk119** [comment]
@flinkbot run azure

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java (hunk 6 above)
Comment (swuferhong) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/Executor.java:65]: Ok, I got it.

**Review by swuferhong — APPROVED** [review]
LGTM +1

**Pull request merged**